ktor-network/jvm/test/io/ktor/network/sockets/tests/UDPSocketTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.sockets.tests
+
+import io.ktor.network.selector.*
+import io.ktor.network.sockets.*
+import io.ktor.util.network.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.debug.junit4.*
+import org.junit.*
+import kotlin.coroutines.*
+import kotlin.io.use
+import kotlin.test.*
+import kotlin.test.Test
+
+class UDPSocketTest : CoroutineScope {
+    private val testJob = Job()
+    private val selector = ActorSelectorManager(Dispatchers.Default + testJob)
+
+    @get:Rule
+    val timeout = CoroutinesTimeout(1000, cancelOnTimeout = true)
+
+    override val coroutineContext: CoroutineContext
+        get() = testJob
+
+    @AfterTest
+    fun tearDown() {
+        testJob.cancel()
+        selector.close()
+    }
+
+    @Test
+    fun testBroadcastFails(): Unit = runBlocking {
+        assertFails {
```

Comment:
I would make the assertion more precise since we also may fail due to UDP bind failure (not enough fds and so on)

Reply:
Thanks for the notice, will fix

Reply:
I saw that the latest commits changed this check to the following:
```
assertFailsWith<SocketException>("Permission denied") { ... }
```
Just so you know, the "Permission denied" string here is only for logging. So it does not actually check the exception message but only the exception type. Is that expected as @cy6erGn0m wanted the assertion to be more precise?

Reply:
The message can be different on each platform